lib/manager/github-actions/extract.ts

```diff
@@ -7,24 +8,42 @@ export function extractPackageFile(content: string): PackageFile | null {
   logger.debug('github-actions.extractPackageFile()');
   const deps: PackageDependency[] = [];
   for (const line of content.split('\n')) {
-    // old github actions syntax will be deprecated on September 30, 2019
-    // after that, the first line can be removed
-    const match =
-      /^\s+uses = "docker:\/\/([^"]+)"\s*$/.exec(line) ||
-      /^\s+uses: docker:\/\/([^"]+)\s*$/.exec(line);
-    if (match) {
-      const [, currentFrom] = match;
+    if (line.trim().startsWith('#')) {
+      continue; // eslint-disable-line no-continue
+    }
+
+    const dockerMatch = /^\s+uses: docker:\/\/([^"]+)\s*$/.exec(line);
+    if (dockerMatch) {
+      const [, currentFrom] = dockerMatch;
       const dep = getDep(currentFrom);
       logger.debug(
         {
           depName: dep.depName,
           currentValue: dep.currentValue,
           currentDigest: dep.currentDigest,
         },
-        'Docker image inside GitHub Actions'
+        'Docker image inside GitHub Workflow'
       );
+      dep.depType = 'docker';
       dep.versioning = dockerVersioning.id;
       deps.push(dep);
+      continue; // eslint-disable-line no-continue
+    }
+
+    const tagMatch = /^\s+-?\s+?uses: (?<depName>[\w-]+\/[\w-]+)(?<path>.*)?@(?<currentValue>.+?)\s*?$/.exec(
+      line
+    );
+    if (tagMatch?.groups) {
+      const { depName, currentValue } = tagMatch.groups;
+      const dep: PackageDependency = {
+        depName,
+        currentValue,
+        commitMessageTopic: '{{depName}}} action',
+        datasource: githubReleasesDatasource.id,
+        versioning: dockerVersioning.id,
```

Comment:
Is this because of a need to support `-compatibility` suffixes?

Reply:
Twas at @viceice's recommendation. Apparently docker is slightly more stringent than `loose` in ways besides it's support for `-compat` suffixes? One other big difference I noted is that docker versioning will reject commit shas, which while technically supported seems correct.

Reply:
First question: in this first implementation are we supporting *tags* and not SHAs? Because if SHA support is to be added later then we can worry about it later too.

Second question: at which stage will we ignore SHAs? e.g. will this extract function skip over them, or will they be included in the list of extract deps?

Reply:
> First question: in this first implementation are we supporting tags and not SHAs? Because if SHA support is to be added later then we can worry about it later too.

My vote is support SHAs in a follow up.

> Second question: at which stage will we ignore SHAs? e.g. will this extract function skip over them, or will they be included in the list of extract deps?

Is there a way that's more efficient? I don't know the answer to this as a `renovatebot` newbie.

Reply:
So one possibility would be the extract function does its own "Does this look like a version?" check and if not then sets a `skipReason` in the dep (e.g. unsupported value). Another is that it passes it through but the lookup process does the check based on the versioning. e.g. semver or docker versioning (whichever we ultimately decide to use here) returns `false` for `isValid()`. We want to make sure that SHAs are handled (ignored) gracefully.

Reply:
Sounds like `skipReason` might be a better ux?

Reply:
@rarkins it's to Support v0 to v1 tag updates. Loose could cause a mixup from v0 to v1.0.1 or  v0.0.1 to v1.

Reply:
Maybe we need a custom versioning for this later, so we can use the compatibility function of versioning to suggest the right update.

Reply:
Switched to extract w/ skip reason. I def think a custom version scheme will be the way forward in the future, to handle the happy path (published tag releases) as well as the less common cases (unpublished tags, commits, or branches).